managers/bosh-manager/BoshTaskStatusPoller.js

```diff
@@ -34,33 +34,77 @@ class BoshTaskStatusPoller extends BaseStatusPoller {
     const instanceId = resourceBody.metadata.name;
     const options = _.get(resourceBody, 'spec.options');
     return DirectorService.createInstance(instanceId, options)
-      .then(directorService => directorService.lastOperation(_.get(resourceBody, 'status.response')))
-      .tap(lastOperationValue => logger.debug('last operation value is ', lastOperationValue))
-      .tap(lastOperationValue => lastOperationOfInstance = lastOperationValue)
-      .then(lastOperationValue => Promise.all([eventmesh.apiServerClient.updateResource({
-        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
-        resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
-        resourceId: instanceId,
-        status: {
-          lastOperation: lastOperationValue,
-          state: lastOperationValue.resourceState
-        }
-      }), Promise.try(() => {
-        if (_.includes([CONST.APISERVER.RESOURCE_STATE.SUCCEEDED, CONST.APISERVER.RESOURCE_STATE.FAILED], lastOperationValue.resourceState)) {
-          // cancel the poller and clear the array
+      .then(directorService => directorService.lastOperation(_.get(resourceBody, 'status.response'))
+        .tap(lastOperationValue => logger.debug('last operation value is ', lastOperationValue))
+        .tap(lastOperationValue => lastOperationOfInstance = lastOperationValue)
+        .then(lastOperationValue => Promise.all([
+          Promise.try(() => {
+            if (lastOperationValue.resourceState === CONST.APISERVER.RESOURCE_STATE.SUCCEEDED &&
+              (lastOperationValue.type === 'create' || lastOperationValue.type === 'update')) {
+              return directorService.director.getDeploymentNameForInstanceId(directorService.guid)
+                .then(deploymentName => directorService.director.getDeploymentIpsFromDirector(deploymentName))
```

Comment:
question : what would happen if BOSH director is unavailable during this call ? Shouldn't the exception be caught here and we proceed with rest of the flow ?

Reply:
If BOSH director is unavailable `lastOperation` itself would fail.
But event then I think we should catch the exception and then go through with operation.